readthedocs/core/unresolver.py

```diff
@@ -44,6 +44,33 @@ class InvalidCustomDomainError(DomainError):
     pass
 
 
+class VersionNotFoundError(UnresolverError):
```

Comment:
Does it make sense to create a `core/unresolver/` application/directory and move these to `core/unresolver/exceptions.py`? It seems the unresolver logic is growing considerably and maybe we are at the point where it makes sense making it module.